### DISCUSSION TITLE
Setting up to develop and encountering issue running build.dart

### DISCUSSION THREAD

**Discussion by ImminentOrbit** [thunder-app/thunder#439] [Q&A]
I've read through the setup section and have followed the steps listed. Trying to run the scripts/build.dart in Android Studio, and also via command line, gives me this output:

E:/Programming/flutter/bin/cache/dart-sdk/bin/dart.exe --enable-asserts C:\Users\Orbit\AndroidStudioProjects\thunder\scripts\build.dart
Checking if Flutter is installed...
Unhandled exception:
ProcessException: The system cannot find the file specified.

  Command: flutter --version
#0      _ProcessImpl._runAndWait (dart:io-patch/process_patch.dart:487:7)
#1      _runNonInteractiveProcessSync (dart:io-patch/process_patch.dart:632:18)
#2      Process.runSync (dart:io-patch/process_patch.dart:68:12)
#3      buildRelease (file:///C:/Users/Orbit/AndroidStudioProjects/thunder/scripts/build.dart:9:41)
#4      main (file:///C:/Users/Orbit/AndroidStudioProjects/thunder/scripts/build.dart:120:3)
#5      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:291:19)
#6      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)

Process finished with exit code 255

It is saying it cannot find a file, but I'm not sure which file that is? Any help?

**Comment by micahmo** [comment]
@Arphahat You can tell which file it's having trouble finding by the output of the command.

``` 
Command: flutter --version
```

It's having trouble finding the `flutter` binary. First, make sure it's in your path. On Windows, I found I also had to update the build script with `flutter.bat` (and I had to change all the other binary names, too).

P.S. You can use the VS Code extensions for Dart/Flutter and generally avoid the build script (if you just want to debug and not create a release deployment).

**Reply by ImminentOrbit** [reply]
Thanks, updating it to use flutter.bat addressed that issue. Looks like there is a new problem trying to run "cider" which I don't think was referenced in the install instructions.

**Reply by micahmo** [reply]
You might have to run `flutter pub global activate cider` to install `cider`. And you may also have to change it to `cider.bat` in the script.

**Reply by hjiangsu** [reply]
As @micahmo mentioned, you don't have to run the build script unless you want to generate a release APK/IPA file

If you want to set up your environment for development purposes, usually VSCode's Flutter/Dart extensions will allow you to do that. If you want quicker help, feel free to join in the Matrix space and we can all discuss it there too!

https://matrix.to/#/#thunderapp:matrix.org
The development room is here: https://matrix.to/#/#development_thunder:matrix.org

**Reply by ImminentOrbit** [reply]
Thanks. It looks to be building, even though it now seems to be unnecessary. I'll take a look at the matrix link.
